frontend/src/components/common/Popup.tsx

```diff
@@ -0,0 +1,20 @@
+import { MouseEventHandler, ReactChild, ReactFragment, ReactPortal } from 'react'
+
+const Popup = (props: {
```

Comment:
Prefer props destructuring with a separate type for props, e.g.

`type PopupProps = {
 handleClose?: MouseEventHandler<HTMLSpanElement>
  children: ReactNode
}

const Popup = ({ context, handleClose }) => {...}`

When a component renders components nested inside it, the convention is to call the prop exactly `children`.  Also, `ReactNode` from 'react' should be the only type you need here. Also when directly destructuring props in the component function arguments, you don't need to write `props.something` everywhere or destructure them manually like `const { prop1, prop2 } = props`.